legacy-sdk/whirlpool/src/impl/whirlpool-impl.ts

```diff
@@ -599,14 +598,18 @@ export class WhirlpoolImpl implements Whirlpool {
     const shouldDecreaseLiquidity = positionData.liquidity.gtn(0);
 
     const rewardsToCollect = this.data.rewardInfos
-      .filter((_, i) => {
+      .map((info, i) => ({ info, index: i }))
+      .filter(({ info, index }) => {
```

Comment:
looks like we are mapping to get the index to provide filter with the index. Does the filter's index paramter help in this case?

Reply:
Also fixed